src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs

```diff
@@ -85,6 +85,13 @@ public HttpConnectionPool(HttpConnectionPoolManager poolManager, HttpConnectionK
                     Debug.Assert(proxyUri != null);
                     break;
 
+                case HttpConnectionKind.ProxyTunnel:
```

Comment:
Why add this?  Why not just use HttpConnectionKind.ProxyConnect?

Reply:
Because it serves different purpose.

In `ProxyTunnel` and `SslProxyTunnel` state, we determine that we want to send a CONNECT request to the proxy server to establish a tunnel (Create a CONNECT `HttpRequestMessage`). Then transmit to `ProxyConnect` state.
In `ProxyConnect` state, we will start the actual connection work, and initiate the connection.